doubly_linked_list/doubly_linked_list.py

```diff
@@ -38,27 +38,55 @@ class DoublyLinkedList:
   def __init__(self, node=None):
     self.head = node
     self.tail = node
+    self.max = 0
 
   def add_to_head(self, value):
-    pass
+    if value > self.max:
+      self.max = value
+    new_link = ListNode(value)
+    if self.head == None:
+      self.head = new_link
+    else:
+      self.head.insert_before(value)
+      self.head = self.head.prev
 
   def remove_from_head(self):
+    if self.head == None:
+      return None
+    old_head = self.head
+    self.head = self.head.prev
+    old_head.delete()
+
     pass
 
   def add_to_tail(self, value):
-    pass
+    if value > self.max:
+      self.max = value
+    new_link = ListNode(value)
+    if self.tail == None:
```

Comment:
same concern as with add_to_head, just reversed: you're only setting the tail here